File: WoofAndroid/data/datasource/src/testFixtures/kotlin/com/perrystreet/woof/datasource/messages/MessagesDataSourceFactory.kt
package com.perrystreet.woof.datasource.messages

import org.koin.test.KoinTest
import org.koin.test.inject

class MessagesDataSourceFactory : KoinTest {
    private val dataSource: FakeMessagesDataSource by inject()

    fun withSendMessageError() = apply {
        dataSource.sendMessageError = IllegalStateException("Could not send message")
    }
}

File: WoofIOS/SwiftPackages/Data/DataSource/Sources/DataSourceFakes/Messages/MessagesDataSourceFactory.swift
import DataSource
import Swinject
import SwinjectAutoregistration

public final class MessagesDataSourceFactory {
    private let dataSource: FakeMessagesDataSource

    public init(_ container: Container) {
        dataSource = container~>
    }

    @discardableResult
    public func withSendMessageError() -> Self {
        dataSource.sendMessageError = .unavailable("Could not send message")
        return self
    }
}
